lib/workload/stateless/postgres_manager/construct/postgresManager.ts

```diff
@@ -0,0 +1,127 @@
+import { Duration } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as nodejs from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as ssm from 'aws-cdk-lib/aws-ssm';
+import { DbAuthType, MicroserviceConfig } from '../function/utils';
+
+export type PostgresManagerConfig = {
+  masterSecretName: string;
+  dbClusterIdentifier: string;
+  microserviceDbConfig: MicroserviceConfig;
+  clusterResourceIdParameterName: string;
+};
+
+export type PostgresManagerStackProps = PostgresManagerConfig & {
+  vpc: ec2.IVpc;
+  lambdaSecurityGroup: ec2.ISecurityGroup;
+};
+
+export class PostgresManager extends Construct {
+  constructor(scope: Construct, id: string, props: PostgresManagerStackProps) {
+    super(scope, id);
+
+    const { dbClusterIdentifier, microserviceDbConfig } = props;
+
+    const masterSecret = secretsmanager.Secret.fromSecretNameV2(
+      this,
+      'RdsMasterSecret',
+      props.masterSecretName
+    );
+
+    const dbClusterResourceId = ssm.StringParameter.valueFromLookup(
+      this,
+      '/orcabus/db-cluster-resource-id'
+    );
+
+    // Let lambda access secret manager via aws managed lambda extension
+    // Ref:
+    // https://aws.amazon.com/blogs/compute/using-the-aws-parameter-and-secrets-lambda-extension-to-cache-parameters-and-secrets/
+    const lambdaLayerGetSecretExtension = lambda.LayerVersion.fromLayerVersionArn(
+      this,
+      'GetSecretExtensionLayer',
+      'arn:aws:lambda:ap-southeast-2:665172237481:layer:AWS-Parameters-and-Secrets-Lambda-Extension-Arm64:11'
```

Comment:
this feels brittle - I mean I realise there isn't much chance we want to spin up the stack in another region - but still is a bit of "magic string".

Of all the lambdas we make - aren't these ones going to be executed like 10 times total a month? Is it really worth optimising the cacheing of the secrets?

Reply:
Yeah fair enough, could make this the normal sdk secret fetch then